src/pages/workspace/members/useShouldShowChangeOwnerPage.ts

```diff
@@ -0,0 +1,34 @@
+import {useEffect, useState} from 'react';
+import type {OnyxEntry} from 'react-native-onyx';
+import {usePersonalDetails} from '@components/OnyxListItemProvider';
+import useCurrentUserPersonalDetails from '@hooks/useCurrentUserPersonalDetails';
+import CONST from '@src/CONST';
+import type {Policy} from '@src/types/onyx';
+
+const useShouldShowChangeOwnerPage = (policy: OnyxEntry<Policy>, accountID: number) => {
+    const personalDetails = usePersonalDetails();
+    const currentUserPersonalDetails = useCurrentUserPersonalDetails();
+    const [shouldShow, setShouldShow] = useState(false);
+
+    useEffect(() => {
+        const isCurrentUserOwner = policy?.owner === currentUserPersonalDetails?.login;
+        const isCurrentUserAdmin = policy?.employeeList?.[personalDetails?.[currentUserPersonalDetails?.accountID]?.login ?? '']?.role === CONST.POLICY.ROLE.ADMIN;
+        const isAccountOwner = policy?.owner === personalDetails?.[accountID]?.login;
+        const isSuccessOrFailed = (!policy?.errorFields && !!policy?.isChangeOwnerFailed) || (!policy?.errorFields?.changeOwner && !!policy?.isChangeOwnerSuccessful);
+        setShouldShow((!isCurrentUserOwner && isCurrentUserAdmin && isAccountOwner) || isSuccessOrFailed);
+    }, [
+        accountID,
+        currentUserPersonalDetails?.accountID,
+        currentUserPersonalDetails?.login,
+        personalDetails,
+        policy?.employeeList,
+        policy?.errorFields,
+        policy?.isChangeOwnerFailed,
+        policy?.isChangeOwnerSuccessful,
+        policy?.owner,
+    ]);
+
+    return shouldShow;
+};
```

Comment:
This feels over-engineered. We should not need a hook for this and we should just check  `policy?.employeeList`

Reply:
> isCurrentUserAdmin is included in WorkspaceMemberDetailsPage

It's not used for the not found view logic.

> For isSuccessOrFailed check, when removing that, we'll show not found page after success transferring

Why would this show the not found view after a successful transfer? I don't think that flow would affect the member existence

Reply:
I don't think the not found view is the correct approach if you are trying to transfer ownership to (or as) an ineligible account. In these cases the page should load normally and the API request will just fail normally. This will still have the go back button function as expected so that's all good.

Let's show the not found view only if the account does not exist (regardless of its permission and regardless of your permission)

Reply:
Actually we should also show the not found view if you don't have the permission (to be consist with other pages).

```js
const shouldShowNotFoundPage = !isCurrentUserOwner || !member;
```

Without the hook and without a state. This should be a derived value.

Reply:
Why is the screen empty?